### ISSUE TITLE
isInStoryPageSideEdge_ incorrectly returns true if `clicked` by pressing enter

### ISSUE THREAD

**Issue by processprocess** [ampproject/amphtml#31611]
"clicking" a focused button by pressing enter or double tapping with talkback with send an event with coordinates originating from 0, 0.

`isInStoryPageSideEdge_` will return `true` and navigate back as a result.

Test by opening [page 4 of this story](https://www.gstatic.com/amphtml/stamp/qa/navigation.html#page=page4) pressing tab until `link1` is selected, then press enter.

We could read the location of the event target instead of the click.
Or we could make an exception if the click originates from 0, 0.

Discovered when testing a11y of #31552.
Screenshot of click origin when tabbing, clicking the `!` button:
<img width="467" alt="Screen Shot 2020-12-15 at 2 03 28 PM" src="https://user-images.githubusercontent.com/3860311/102260340-508cd200-3ede-11eb-9d05-1ea5c0c57d94.png">

This bug is related to [30183](https://github.com/ampproject/amphtml/issues/30183)

cc @ampproject/wg-stories

**Comment by processprocess** [comment]
Once this is fixed it needs to be added as an a11y test in the manual QA.

**Comment by gmajoulet** [comment]
Great catch!!

Making an exception for the (0, 0) case would be much easier than looking at the event target position, which would be a breaking change from how things work today, where an element can partially be in the protected zone but still be clickable.

**Comment by raxsha** [comment]
Added the following manual QA checks:

- Tapping a link in the left or right 48px of a story page should navigate within the story, not open the link. Please verify on desktop and mobile.
- **A11y exception: Clicking links with Talkback (by focusing them with tab and clicking enter) should always result in link navigation and never page navigation. Links should be tested with mouse clicks as well as keyboard "clicks" to account for these cases.**
- Tapping a link a link that occupies over 80% of a story page should be ignored since it is navigation blocking.
- **A11y exception: Clicking links with Talkback (by focusing them with tab and clicking enter) should always result in link navigation and never page navigation. Links should be tested with mouse clicks as well as keyboard "clicks" to account for these cases.**

**Comment by raxsha** [comment]
**Note**: this was actually NOT added to our manual QA testing since a11y testing is not currently included in our instructions to testers and more work is needed to start a11y testing thoroughly.
